server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -525,6 +563,23 @@ private UploadedMetadataResults writeMetadataInParallel(
             uploadTasks.put(indexMetadata.getIndex().getName(), getIndexMetadataAsyncAction(clusterState, indexMetadata, listener));
         });
 
+        indicesRoutingToUpload.forEach(indexRoutingTable -> {
+            try {
+                uploadTasks.put(
+                    RemoteRoutingTableService.INDEX_ROUTING_METADATA_PREFIX + indexRoutingTable.getIndex().getName(),
+                    remoteRoutingTableService.get()
```

Comment:
We do not here have a check on presence of `remoteRoutingTableService ` can we run into NPE?

Reply:
We are creating `indicesRoutingToUpload` only if `remoteRoutingTableService` is present, so this condition will not occur